src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java

```diff
@@ -23,43 +24,45 @@
      * Overrides the authorship information in {@code fileInfo} based on annotations given on the file.
      */
     public static void aggregateAnnotationAuthorInfo(FileInfo fileInfo, Map<String, Author> authorAliasMap) {
-        Author currentAuthor = Author.UNKNOWN_AUTHOR;
+        Optional<Author> currentAnnotatedAuthor = Optional.empty();
         Path filePath = Paths.get(fileInfo.getPath());
         for (LineInfo lineInfo : fileInfo.getLines()) {
             if (lineInfo.getContent().contains(AUTHOR_TAG)) {
-                Author newAuthor = findAuthorInLine(lineInfo.getContent(), authorAliasMap);
+                Optional<Author> newAnnotatedAuthor = findAuthorInLine(lineInfo.getContent(), authorAliasMap);
 
-                if (newAuthor.equals(Author.UNKNOWN_AUTHOR)) {
-                    //end of an author tag should belong to this author too.
-                    lineInfo.setAuthor(currentAuthor);
-                } else if (newAuthor.getIgnoreGlobMatcher().matches(filePath)) {
-                    newAuthor = Author.UNKNOWN_AUTHOR;
+                if (!newAnnotatedAuthor.isPresent()) {
+                    // end of an author tag should belong to the current author too.
+                    // if an end author tag was used without a corresponding starting tag, attribute the
+                    // line to UNKNOWN_AUTHOR
+                    lineInfo.setAuthor(currentAnnotatedAuthor.orElse(Author.UNKNOWN_AUTHOR));
```

Comment:
I don't quite get this particular comment; can you give an example for this part?

Reply:
The `orElse` caters for cases where the user used an end author tag (`@@author`) without having any preceding start author tag (e.g. `@@author tom`). This is most likely because the user forgot to provide a start author tag.

In such cases, we will attribute the line with the end author tag to `UNKNOWN_AUTHOR`

This logic is consistent with the current implementation:
https://github.com/reposense/RepoSense/blob/9725a25d4e34164cefbb51f7f6e39964cb0a521a/src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java#L25-L34

Reply:
I see. Thanks